### DISCUSSION TITLE
An hexagon autoscheduler problems about halide.

### DISCUSSION THREAD

**Discussion by pengxuan2022** [halide/Halide#7001] [Q&A] [answered]
When I directly use the conv3x3 code automatically generated by halide 13 autoscheduler in offload mode(target=arm-64-linux-hvx-hvx_128), the vector length is always wrong(use arm cpu vector length). However, the vector length generated by halide software using Qualcomm's official SDK is correct and has better performance. 
When target is hexagon-32-qurt-hvx_128, the vector length generated by halide 13 autoscheduler is right .
Is the halide software of Qualcomm SDK modifying the original code? Or is the latest code not uploaded?

**Comment by abadams** [comment]
I don't believe any of the Halide autoschedulers are trained for HVX, so you should not expect them to work. Using Halide with HVX requires writing manual schedules, and I believe that's what the Qualcomm SDK does.

**Comment by pranavb-ca** [comment, marked as answer]
Hi @pengxuan2022 - depending on the version of the Qualcomm SDK you have we do have an experimental autoscheduler which will use the right vector length. We believe it still needs some work before it is in a state worthy of submitting upstream - at the moment it is a little hacky. Like Andrew said, the community autoscheduler will not schedule for Hexagon.
